### PR TITLE
ENH: Creating and registering VR data module widget

### PR DESCRIPTION
**Pull request by mohammadrashid0917**
Creates and registers the VR data module widget (which can be switched to from the VR home widget)

### CODE REVIEW

**Comment by cpinter** [comment]
Thanks, Mohammad!

File: VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.cxx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  return (d->VirtualRealityViewNode.GetPointer() ? d->VirtualRealityViewNode->GetID() : QString());
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityDataModuleWidget::updateWidgetFromMRML()
+{
+  //Q_D(qMRMLVirtualRealityDataModuleWidget);
+}
+
+qMRMLSubjectHierarchyTreeView* qMRMLVirtualRealityDataModuleWidget::treeView()
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.cxx]: Please add separator that appears above each function.

Also please add an empty new line to the end of the file

File: VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#include "qMRMLSubjectHierarchyModel.h"
+
+class vtkMRMLVirtualRealityViewNode;
+class qMRMLVirtualRealityDataModuleWidgetPrivate;
+
+/// \ingroup SlicerVirtualReality_Widgets
+class Q_SLICER_QTMODULES_VIRTUALREALITY_WIDGETS_EXPORT qMRMLVirtualRealityDataModuleWidget : public qMRMLWidget
+{
+  Q_OBJECT
+  QVTK_OBJECT
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.h]: I believe we don't need the QVTK_OBJECT declaration. Please remove it and try to build it to confirm. Thanks!

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 3]
```diff
@@ -53,6 +53,7 @@ class qMRMLVirtualRealityHomeWidgetPrivate : public Ui_qMRMLVirtualRealityHomeWi
 public:
   /// Virtual reality view MRML node
   vtkWeakPointer<vtkMRMLVirtualRealityViewNode> VirtualRealityViewNode;
+  qMRMLVirtualRealityDataModuleWidget* DataModuleWidget = new qMRMLVirtualRealityDataModuleWidget;
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx]: You cannot do this. Only declare here, create in constructor.

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 4]
```diff
@@ -174,10 +183,13 @@ void  qMRMLVirtualRealityHomeWidget::onBackButtonPressed()
 }
 
 //-----------------------------------------------------------------------------
-QString qMRMLVirtualRealityHomeWidget::virtualRealityViewNodeID()const
+void qMRMLVirtualRealityHomeWidget::registerDataModule()
 {
-  Q_D(const qMRMLVirtualRealityHomeWidget);
-  return (d->VirtualRealityViewNode.GetPointer() ? d->VirtualRealityViewNode->GetID() : QString());
+  Q_D(qMRMLVirtualRealityHomeWidget);
+  QIcon dataIcon(QPixmap("C:/D/S4/Modules/Loadable/Data/Resources/Icons/SubjectHierarchy.png"));
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx]: No computer-specific local paths please!!!

File: VirtualReality/Widgets/qMRMLVirtualRealityView.cxx [hunk 5]
```diff
@@ -620,6 +621,7 @@ qMRMLVirtualRealityView::qMRMLVirtualRealityView(QWidget* _parent) : Superclass(
 {
   Q_D(qMRMLVirtualRealityView);
   d->init();
+
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityView.cxx]: Remove empty line from here

File: VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.h (hunk 2 above)
Comment (mohammadrashid0917) [reply VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.h]: The build failed when I removed QVTK_OBJECT, I got this error: 
C3861 'qvtkReconnect': identifier not found

Comment (cpinter) [reply VirtualReality/Widgets/qMRMLVirtualRealityDataModuleWidget.h]: Thanks for checking! I didn't notice the vtkReconnect call. Please add it back

**Pull request closed without merging**